src/validation.cpp

```diff
@@ -3530,6 +3529,7 @@ bool CChainState::AcceptBlock(const std::shared_ptr<const CBlock>& pblock, CVali
         GetMainSignals().NewPoWValidBlock(pindex, pblock);
 
     // Write block to history file
+    if (fNewBlock) *fNewBlock = true;
```

Comment:
Sorry, can't this be right before `return true`?

Reply:
Yes, this could be anywhere between this line and the line you mentioned. Though, since this went through more than a week of bike-shedding, I'd prefer if we just fixed the bug and then followed up with nits in later pull requests.

Reply:
Is it a nit? Is it a new block if anything fails below?

Reply:
If anything below fails, the node is supposed to shut down immediately